src/main/java/com/commercetools/service/ctp/impl/CartServiceImpl.java

```diff
@@ -26,16 +28,35 @@ public CartServiceImpl(@Nonnull SphereClient sphereClient) {
 
     @Override
     public CompletionStage<Optional<Cart>> getByPaymentId(@Nullable String paymentId) {
+        return getByPaymentId(paymentId, null);
+    }
+
+    @Override
+    public CompletionStage<Optional<Cart>> getByPaymentId(@Nullable String paymentId,
+                                                          @Nullable String expansionPath) {
```

Comment:
we should follow JVM SDK native declarative style as match as possible, so it is strongly recommended to have this function with Object/Functional-oriented style, instead of String-expansion style. Thus this method should be declared/implemented as:

```java
@Override
public CompletionStage<Optional<Cart>> getByPaymentId(@Nullable String paymentId,
                                                      @Nullable Function<CartExpansionModel<Cart>, ExpansionPathContainer<Cart>> cartExpansionFunction) {
    if (StringUtils.isEmpty(paymentId)) {
        return completedFuture(Optional.empty());
    }
    CartQuery cartQuery = CartQuery.of()
            .withPredicates(m -> m.paymentInfo().payments().id().is(paymentId));
    if (cartExpansionFunction != null) {
        cartQuery.plusExpansionPaths(cartExpansionFunction);
    }
    return sphereClient.execute(cartQuery).thenApplyAsync(PagedQueryResult::head);
}
```

and then call it as:

```java
getByPaymentId(paymentId, expansionModel -> expansionModel.paymentInfo().payments())
```

Also same for [`getByPaymentMethodAndInterfaceId()`](https://github.com/commercetools/commercetools-paypal-plus-integration/pull/49/files#diff-543f69643509f1897369914ca1793923R59) below

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/49/commits/b3cf8e3e026c525bad5bfc997e6dba02f42d6026